src/sqlfluff/dialects/dialect_mysql.py

```diff
@@ -1114,6 +1114,20 @@ class SetAssignmentStatementSegment(BaseSegment):
 
     match_grammar = Sequence(
         "SET",
+        Sequence(
+            Delimited(
+                Ref("SetStatementElementSegment"), terminator=Ref("SemicolonSegment")
+            )
+        ),
+    )
+
+
+class SetStatementElementSegment(BaseSegment):
```

Comment:
I'm in two minds as to whether you need a separate class for this or it it should all be in the `SetAssignmentStatementSegment`.

On reflection, I think keep it in there, to match Snowflake.

WDYT?

Reply:
Initially I implemented everything in one class. But in #3328 on the expected parse tree, there is set_statement_element twice, so I tried to replicate it as closely as possible.

Reply:
Could you show me that code? Cause I just tried this and it seems to work:

```py
class SetAssignmentStatementSegment(BaseSegment):
    """A `SET` statement.

    https://dev.mysql.com/doc/refman/8.0/en/set-variable.html
    """

    type = "set_statement"

    match_grammar = Sequence(
        "SET",
        Delimited(
            Sequence(
                OneOf(Ref("SessionVariableNameSegment"), Ref("LocalVariableNameSegment")),
                Ref("EqualsSegment"),
                AnyNumberOf(
                    Ref("QuotedLiteralSegment"),
                    Ref("DoubleQuotedLiteralSegment"),
                    Ref("SessionVariableNameSegment"),
                    # Match boolean keywords before local variables.
                    Ref("BooleanDynamicSystemVariablesGrammar"),
                    Ref("LocalVariableNameSegment"),
                    Ref("FunctionSegment"),
                    Ref("ArithmeticBinaryOperatorGrammar"),
                ),
            ),
        ),
    )
```

Reply:
The expected parse tree on the issue was:
```
% echo "SET a = 1, b = 2;" | sqlfluff parse - --dialect mysql
|file:
|    statement:
|        set_statement:
|            keyword:                                          'SET'
|            whitespace:                                       ' '
|            set_statement_element:
|                variable:                                     'a'
|                whitespace:                                   ' '
|                comparison_operator:
|                    raw_comparison_operator:                  '='
|                whitespace:                                   ' '
|                variable:                                     '1'
|            comma:                                            ','
|            whitespace:                                       ' '
|            set_statement_element:
|                variable:                                     'b'
|                whitespace:                                   ' '
|                comparison_operator:
|                    raw_comparison_operator:                  '='
|                whitespace:                                   ' '
|                variable:                                     '2'
|    statement_terminator:                                     ';'
|    newline:                                                  '\n'
```
I though because the set_statement_element was being reused twice (or more times if there are more variables), it should be on it's own class. However it definitely works this way as well.

Reply:
There's quite a few instances where delimited items are in not in their own class.

It depends on whether we want it, in its own part of the parse tree or not.

We don't currently have any use cases that necessitate that. That's not to say there won't be any in future.

My argument to keep it in the `set_statement` class is for consistency with the Snowflake dialect which also has its own `set_statement` and does keep it in that class. We should try to keep similar statements defined similarly, otherwise it can cause assumptions, misunderstandings and bugs when writing rules or similar.

It would also be a simpler change, and have a lot less YAML changes in this PR.

However, I could be persuaded to separate them out into their own class IF we also updated Snowflake similarly.

Reply:
Sure I agree, we should probably keep it in the same class for consistency and simplicity. I only did it to match the expected parse tree, but it doesn't make much sense honestly. We could always pull it out in a new class if we need it in the future.

Reply:
@jivanpal was there a specific reason you wanted these in `set_statement_element` blocks? Or are you happy if it parses without them?

Reply:
@tunetheweb No specific need for it, I just adapted the likes of `SELECT 1, 2;`, in which `1` and `2` are each a `select_clause_element`.

Reply:
OK @cgkoutzigiannis let's do without it for now, to make SnowFlake's implementation.